packages/studio-base/src/panels/ThreeDeeRender/renderables/primitives/RenderableTexts.ts

```diff
@@ -0,0 +1,127 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import { toNanoSec } from "@foxglove/rostime";
+import { SceneEntity, TextPrimitive } from "@foxglove/schemas/schemas/typescript";
+import { emptyPose } from "@foxglove/studio-base/util/Pose";
+import { Label, LabelPool } from "@foxglove/three-text";
+
+import type { Renderer } from "../../Renderer";
+import { getLuminance, makeRgba, SRGBToLinear, stringToRgba } from "../../color";
+import { LayerSettingsEntity } from "../SceneEntities";
+import { RenderablePrimitive } from "./RenderablePrimitive";
+
+const tempRgba = makeRgba();
+
+export class RenderableTexts extends RenderablePrimitive {
+  private labelPool: LabelPool;
+  private labels: Label[] = [];
+
+  public constructor(renderer: Renderer) {
+    super("", renderer, {
+      receiveTime: -1n,
+      messageTime: -1n,
+      frameId: "",
+      pose: emptyPose(),
+      settings: { visible: true, color: undefined },
+      settingsPath: [],
+      entity: undefined,
+    });
+
+    this.labelPool = renderer.labelPool;
+  }
+  private _ensureCapacity(newLength: number): void {
+    const oldLength = this.labels.length;
+    if (newLength > oldLength) {
+      for (let i = oldLength; i < newLength; i++) {
+        const newLabel = this.labelPool.acquire();
+        this.labels.push(newLabel);
+        this.add(newLabel);
+      }
+    }
+  }
+
+  private _updateMesh(texts: TextPrimitive[]) {
```

Comment:
```suggestion
  private _updateTexts(texts: TextPrimitive[]) {
```

I regret naming this method something different in all of the renderables 🤣